### DISCUSSION TITLE
Using settings in services

### DISCUSSION THREAD

**Discussion by Dezmonter** [spatie/laravel-settings#234] [Q&A]
I am using your wonderful package but I got an error. I have a Language service and a LanguageServiceProvider that is called during migration (boot() method). The Language service requires settings from your package and I get the error Table 'XXX.settings' doesn't exist(((

![image](https://github.com/spatie/laravel-settings/assets/46536362/1b4e73da-9749-459c-af79-39721c27e9bb)

![image](https://github.com/spatie/laravel-settings/assets/46536362/99ba8256-abb4-4e07-84b3-1c4590443e06)

how can i solve my problem ?

**Comment by rubenvanassche** [comment]
Looks like your database isn't booted at that this point, injecting it somewhere later on when everything is booted is the correct way. Sometimes using `app()` or `resolve()` is required for this.
